arm/Microsoft.OperationalInsights/workspaces/deploy.bicep

```diff
@@ -0,0 +1,844 @@
+  properties: {
+    objectName: 'Network Interface'
+    instanceName: '*'
+    intervalSeconds: 60
+    counterName: 'Packets Received/sec'
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_windowsPerfCounter45 'datasources@2020-03-01-preview' = {
+  name: 'windowsPerfCounter45'
+  kind: 'WindowsPerformanceCounter'
+  properties: {
+    objectName: 'Network Interface'
+    instanceName: '*'
+    intervalSeconds: 60
+    counterName: 'Packets Outbound Errors'
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_windowsPerfCounter46 'datasources@2020-03-01-preview' = {
+  name: 'windowsPerfCounter46'
+  kind: 'WindowsPerformanceCounter'
+  properties: {
+    objectName: 'Network Interface'
+    instanceName: '*'
+    intervalSeconds: 60
+    counterName: 'Packets Received Errors'
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_sampleIISLog1 'datasources@2020-03-01-preview' = if (false) {
+  name: 'sampleIISLog1'
+  kind: 'IISLogs'
+  properties: {
+    state: 'OnPremiseEnabled'
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_sampleSyslog1 'datasources@2020-03-01-preview' = {
+  name: 'sampleSyslog1'
+  kind: 'LinuxSyslog'
+  properties: {
+    syslogName: 'kern'
+    syslogSeverities: [
+      {
+        severity: 'emerg'
+      }
+      {
+        severity: 'alert'
+      }
+      {
+        severity: 'crit'
+      }
+      {
+        severity: 'err'
+      }
+      {
+        severity: 'warning'
+      }
+    ]
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_sampleSyslogCollection1 'datasources@2020-03-01-preview' = {
+  name: 'sampleSyslogCollection1'
+  kind: 'LinuxSyslogCollection'
+  properties: {
+    state: 'Enabled'
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_sampleLinuxPerf1 'datasources@2020-03-01-preview' = {
+  name: 'sampleLinuxPerf1'
+  kind: 'LinuxPerformanceObject'
+  properties: {
+    performanceCounters: [
+      {
+        counterName: '% Used Inodes'
+      }
+      {
+        counterName: 'Free Megabytes'
+      }
+      {
+        counterName: '% Used Space'
+      }
+      {
+        counterName: 'Disk Transfers/sec'
+      }
+      {
+        counterName: 'Disk Reads/sec'
+      }
+      {
+        counterName: 'Disk Writes/sec'
+      }
+    ]
+    objectName: 'Logical Disk'
+    instanceName: '*'
+    intervalSeconds: 10
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_sampleLinuxPerfCollection1 'datasources@2020-03-01-preview' = {
+  name: 'sampleLinuxPerfCollection1'
+  kind: 'LinuxPerformanceCollection'
+  properties: {
+    state: 'Enabled'
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_AzureActivityLog 'datasources@2020-03-01-preview' = [for item in activityLogAdditionalSubscriptionIDs: {
+  kind: 'AzureActivityLog'
+  name: '${(empty(activityLogAdditionalSubscriptionIDs) ? 'placeholder' : item)}'
+  properties: {
+    linkedResourceId: '/subscriptions/${item}/providers/microsoft.insights/eventTypes/management'
+  }
+  dependsOn: [
+    logAnalyticsWorkspace
+  ]
+}]
```

Comment:
I'd vote for parameterizing this as much as possible and have a module iterating on datasourcers resources

Reply:
Agreed. But let's do that in a separate PR as this seems to be a bigger change (to me)